### PR TITLE
feat(Datagrid): added select all button to customize columns modal; added styles; revised instruction text

### PR DESCRIPTION
**Pull request by kai-liu**
Contributes to
[https://github.com/carbon-design-system/ibm-cloud-cognitive/issues/1613](https://github.com/carbon-design-system/ibm-cloud-cognitive/pull/url)
[https://github.com/carbon-design-system/ibm-cloud-cognitive/issues/1981](https://github.com/carbon-design-system/ibm-cloud-cognitive/pull/url)

Implemented select all functionalities to the columns.
Implemented style changes to match the style of a Datagrid table.
Revised text to improve clarity.

What did you change?
within the Datagrid directory:

`DraggableElement.js` Added selected attribute to help with css selectors
`addons/CustomizeColumns/Actions.js` Removed search button, as required by design.

`CustomizeColumns/Columns.js` Added select-all element and implement logic to support it

`CustomizeColumns/CustomizeColumnsModal.js` Added logic to show number of visible columns vs total number of columns. Removed elements in accordance to the carbon design document [https://pages.github.ibm.com/cdai-design/pal/components/data-table/column-customization/usage/](https://github.com/carbon-design-system/ibm-cloud-cognitive/pull/url)

`/styles/addons/CustomizeColumnsModal.scss` Style changes.

`/styles/draggableElement.scss' Style changes

How did you test and verify your work?
storybook

### CODE REVIEW

File: packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Columns.js [hunk 1]
```diff
@@ -90,6 +95,32 @@ const Columns = ({
               ? instructionsLabel
               : disabledInstructionsLabel}
           </span>
+          <div
+            id={'customize-columns-select-all'}
+            className={classNames(`${blockClass}__draggable-handleHolder`, {
+              ['-selected']: getVisibleColumnsCount() > 0,
```
Comment (matthewgallo) [inline packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Columns.js:101]: Can this class change from `-selected` to something like `${blockClass}__draggable-handleHolder--selected`?

File: packages/cloud-cognitive/src/components/Datagrid/styles/_draggableElement.scss [hunk 2]
```diff
@@ -56,3 +72,19 @@ svg.#{$block-class}__draggable-handleStyle.disable {
...
+  background-color: $hover-selected-ui;
+}
+
+.-selected {
+  background-color: $layer-selected;
+}
+
+.-selected:hover {
+  background-color: $hover-selected-ui;
+}
```
Comment (matthewgallo) [inline packages/cloud-cognitive/src/components/Datagrid/styles/_draggableElement.scss:90]: These all need to be scoped within the block class.

```scss
.#{$block-class} [selected='true'] {
...
}
```

File: packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Columns.js (hunk 1 above)
Comment (kai-liu) [reply packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Columns.js:101]: Replaced `className={classNames(`${blockClass}__draggable-handleHolder`, {
              ['-selected']: getVisibleColumnsCount() > 0,`
with `className={classNames({
              [`${blockClass}__draggable-handleHolder`]: getVisibleColumnsCount() === 0,
              [`${blockClass}__draggable-handleHolder-selected`]: getVisibleColumnsCount() > 0,
           })}`

Also adjusted selectors in `_draggableElement.scss` to match the class names

File: packages/cloud-cognitive/src/components/Datagrid/styles/_draggableElement.scss (hunk 2 above)
Comment (kai-liu) [reply packages/cloud-cognitive/src/components/Datagrid/styles/_draggableElement.scss:90]: Reconsolidated selectors. These selectors have been removed to reduce redundancy and improve readability.

**Comment by elycheea** [comment]
@allcontributors add @matthewgallo for review

File: packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Columns.js [hunk 3]
```diff
@@ -7,7 +7,8 @@
  * restricted by GSA ADP Schedule Contract with IBM Corp.
  */
 
-import * as React from 'react';
+// import * as React from 'react';
```
Comment (matthewgallo) [inline packages/cloud-cognitive/src/components/Datagrid/Datagrid/addons/CustomizeColumns/Columns.js:10]: Can you remove this line?

**Review by matthewgallo — APPROVED** [review]
Thanks for all the hard work on this @kai-liu 🎉

**Comment by elycheea** [comment]
@allcontributors add @kai-liu for code

**Pull request merged**